src/services/usersService.js

```diff
@@ -1,18 +1,71 @@
 import api from "@api/api";
 import showCustomAlert from "@utils/show_alert";
 import store from "@store/store";
-import { history } from "@helpers/history";
 
 import validateEmail from "@validations/user/validateEmail";
 import validateFieldLength from "@validations/validateFieldLength";
 import validateIsEmptyObject from "@validations/validateIsEmtpyObject";
+import validateLogin from "@validations/user/validateLogin";
+import validatePassword from "@validations/user/validatePassword";
 import validatePhone from "@validations/user/validatePhone";
 
 class UsersService {
   constructor() {
     this.currentUser = store.getState().currentUser.value;
   }
 
+  async login(data) {
```

Comment:
There is not much sense in this because when have `async` here, it means the method already return promise.
```
  async login(data) {
    return new Promise(async (resolve, reject) => {
```

I simply propose to do the following:

```
async login(data) {
      const { login, password } = data;

      if (!login?.length || !password?.length) {
          throw "The login and password fields must not be empty."
      }
}
```

Reply:
please update it and I will approve